src/runtime/virtcontainers/qemu_amd64_test.go

```diff
@@ -22,26 +22,36 @@ func qemuConfig(machineType string) HypervisorConfig {
 	}
 }
 
-func newTestQemu(machineType string) qemuArch {
+func newTestQemu(assert *assert.Assertions, machineType string) qemuArch {
 	config := qemuConfig(machineType)
-	return newQemuArch(config)
+	arch, err := newQemuArch(config)
+	assert.NoError(err)
+	return arch
+}
+
+func TestQemuAmd64BadMachineType(t *testing.T) {
+	assert := assert.New(t)
+
+	config := qemuConfig("no-such-machine-type")
+	_, err := newQemuArch(config)
+	assert.Error(err)
 }
 
 func TestQemuAmd64Capabilities(t *testing.T) {
 	assert := assert.New(t)
 
-	amd64 := newTestQemu(QemuPC)
+	amd64 := newTestQemu(assert, QemuPC)
 	caps := amd64.capabilities()
 	assert.True(caps.IsBlockDeviceHotplugSupported())
 
-	amd64 = newTestQemu(QemuQ35)
+	amd64 = newTestQemu(assert, QemuQ35)
 	caps = amd64.capabilities()
 	assert.True(caps.IsBlockDeviceHotplugSupported())
 }
 
 func TestQemuAmd64Bridges(t *testing.T) {
 	assert := assert.New(t)
-	amd64 := newTestQemu(QemuPC)
+	amd64 := newTestQemu(assert, QemuPC)
```

Comment:
Could you add a test to check when the machine type is unsupported ?

Reply:
I added TestQemuAmd64BadMachineType in the first commit of the set.  Is that what you're looking for?

Reply:
sorry, I didn't see it, thanks